dts/bindings/sensor/zephyr,sensing-sensor.yaml

```diff
@@ -0,0 +1,34 @@
+# Copyright (c) 2023, Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+description: Sensor subsystem sensor common properties bindings.
+
+include: sensor-device.yaml
+
+properties:
+  sensor-type:
+    type: int
+    required: true
+    description: sensor type id (follow HID spec definition)
+
+  sensor-index:
+    type: int
+    required: true
+    description: sensor instance index for same sensor type
+
+  friendly-name:
+    required: true
+
+  minimal-interval:
+    type: int
+    required: true
+    description: sensor minimal report interval
+
+  rotation-matrix:
+    type: array
+    description: sensor rotation matrix
+
+  reporters:
+    type: phandles
+    description: sensor reporters
```

Comment:
Please refer to this comments: https://github.com/zephyrproject-rtos/zephyr/pull/55389/files#r1205227812
thanks.

Reply:
Per discussion before, this is sensing subsystem design. I am afraid we cannot modify this easily.

Reply:
Please don't unresolve this comment, this is a serious issue with the design which is exactly what I'm trying to point out. Having 2 code paths for doing the exact same thing is not a good design and bloats the memory requirements for the subsystem.

Reply:
I'm not very understand the 2 code paths you mean. 

Internally there is single config path and data path, **Application** and **Sensor** clients as sensor config initiator and sensor data consumer all using single config path and data path, the differences are:
- **Application** client(s) need call **Sensing Subsystem API** explicitly to open sensor(s) 
- **Sensor** client(s) is no needed, **Sensing Subsystem** will auto open reporter sensors to client sensors according device tree via **reporters** property. 

The whole reporting topology between sensors can be statically determined and described by device tree (the **reporters** property is key), so the **Sensing Subsystem** can auto open connections between sensors. But it can't to know which **Application** client(s) will open which sensor(s), it's dynamic, so it required **Application(s)** explicitly calling **Sensing Subsystem API** to open connection(s) to sensor(s). Make sense? 

That's why you can see we use single unified `sensing_sensor_handle` for all connections:  **Application** <-> **Sensor**, and **sensor** <-> **sensor**. Since they use same data path/config path internally. 



We defined two kinds of APIs: **Sensing Subsystem API** for **Application**, and **Sensing Sensor API** for **sensor**, it's not mean internally will have 2 different paths for each. Why need two kinds of APIs, because of the nature of usages:

- **Application** need API to open/config/receive data to/from sensor(s)
  - most of these APIs are active, only some callbacks to receive data/events are passive. 
- sensor vendors need API to implement sensors. 
  - most of these APIs are passive callbacks, for config reporter sensor, can reuse the config API for **Application**, since it uses same **sensing_sensor_handle** as **Application**. That's why I mean internally there is single paths for **Application** and **Sensor**.

This design can keep sensor vendors focus on development of the sensors, applications can focus on how using sensors, **Sensing Subsystem** connects them all and help build a sensing ecosystem, connect various sensors with standard APIs. 
  
For the flow details can see doc: https://github.com/ghu0510/zephyr/blob/ghu_senss_upstream_pr1/doc/services/sensing/index.rst

And the API explanation in https://github.com/zephyrproject-rtos/zephyr/pull/55389/files#r1205227812 

This is the core concepts of the design, and I explained many times, **if it's still not clear, then it's my fault, I can explain more and more** :)

Reply:
@yperess If you agree, please mark it as resolved, if you don't' agree, please put your comments, we can continue the discuss, but if you don't response the replay, we'll mark it as resolved, of course you can reopen or create new one. 

Appreciate for your great review efforts, many of them are very good, but since it's had many comments and discussions, mark the status of them can help we clearly know the whole status. Thanks a lot!

Reply:
not blocking this PR, resolve it. 
Agreed for future discuss in separated issue: #58925